### ISSUE TITLE
keydb-server v5.1.1 crash on keydb-cli config set activedefrag "yes"

### ISSUE THREAD

**Issue by rnz** [JohnSully/KeyDB#131]
After comand 
```
keydb-cli config set activedefrag "yes"
```
KeyDB crash:

```
# keydb-cli info server  
# Server
redis_version:5.1.1
redis_git_sha1:a91f58b8
redis_git_dirty:1
redis_build_id:f8c7776a6b2fa5be
redis_mode:standalone
os:Linux 4.15.18-9-pve x86_64
arch_bits:64
multiplexing_api:epoll
atomicvar_api:atomic-builtin
gcc_version:6.3.0
process_id:1720010
run_id:4b3c2e383b0b0ba955ea04a5fd88ffd88d6ecc4a
tcp_port:6379
uptime_in_seconds:332
uptime_in_days:0
hz:10
configured_hz:10
lru_clock:863189
executable:/usr/local/bin/keydb-server
config_file:/etc/keydb/redis.conf
```

```
=== KEYDB BUG REPORT START: Cut & paste starting from here ===
1679786:M 01 Jan 2020 23:22:59.937 # === ASSERTION FAILED ===
1679786:M 01 Jan 2020 23:22:59.937 # ==> db.cpp:1215 'itr != db->setexpire->end()' is not true
1679786:M 01 Jan 2020 23:22:59.937 # (forcing SIGSEGV to print the bug report.)
1679786:M 01 Jan 2020 23:22:59.937 # KeyDB 5.1.1 crashed by signal: 11
1679786:M 01 Jan 2020 23:22:59.937 # Crashed running the instruction at: 0x5559f109be1e
1679786:M 01 Jan 2020 23:22:59.937 # Accessing address: 0xffffffffffffffff
1679786:M 01 Jan 2020 23:22:59.938 # Failed assertion: itr != db->setexpire->end() (db.cpp:1215)

------ STACK TRACE ------
EIP:
/usr/local/bin/keydb-server 0.0.0.0:6379(_serverAssert+0x7e) [0x5559f109be1e]

Backtrace:
/usr/local/bin/keydb-server 0.0.0.0:6379(logStackTrace(ucontext*)+0x32) [0x5559f109d472]
/usr/local/bin/keydb-server 0.0.0.0:6379(sigsegvHandler(int, siginfo_t*, void*)+0xc8) [0x5559f109db98]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x110e0) [0x7fadc591b0e0]
/usr/local/bin/keydb-server 0.0.0.0:6379(_serverAssert+0x7e) [0x5559f109be1e]
/usr/local/bin/keydb-server 0.0.0.0:6379(removeExpireCore(redisDb*, redisObject*, dictEntry*)+0xdb7) [0x5559f1068c67]
/usr/local/bin/keydb-server 0.0.0.0:6379(dbOverwriteCore(redisDb*, dictEntry*, redisObject*, redisObject*, bool, bool)+0x15d) [0x5559f106a10d]
/usr/local/bin/keydb-server 0.0.0.0:6379(setKey(redisDb*, redisObject*, redisObject*)+0x5b) [0x5559f106a3db]
/usr/local/bin/keydb-server 0.0.0.0:6379(setGenericCommand(client*, int, redisObject*, redisObject*, redisObject*, int, redisObject*, redisObject*)+0x87) [0x5559f107ca27]
/usr/local/bin/keydb-server 0.0.0.0:6379(setCommand(client*)+0x108) [0x5559f107cc88]
/usr/local/bin/keydb-server 0.0.0.0:6379(call(client*, int)+0xc7) [0x5559f1048007]
/usr/local/bin/keydb-server 0.0.0.0:6379(processCommand(client*, int)+0xac7) [0x5559f104c817]
/usr/local/bin/keydb-server 0.0.0.0:6379(processCommandAndResetClient(client*, int)+0x21) [0x5559f1055c01]
/usr/local/bin/keydb-server 0.0.0.0:6379(processInputBuffer(client*, int)+0x7a) [0x5559f105a1fa]
/usr/local/bin/keydb-server 0.0.0.0:6379(readQueryFromClient(aeEventLoop*, int, void*, int)+0x330) [0x5559f105cc00]
/usr/local/bin/keydb-server 0.0.0.0:6379(ProcessEventCore+0x176) [0x5559f1040826]
/usr/local/bin/keydb-server 0.0.0.0:6379(aeProcessEvents+0x145) [0x5559f1040a05]
/usr/local/bin/keydb-server 0.0.0.0:6379(aeMain+0x35) [0x5559f1040d95]
/usr/local/bin/keydb-server 0.0.0.0:6379(workerThreadMain(void*)+0x97) [0x5559f1045117]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x74a4) [0x7fadc59114a4]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7fadc5653d0f]
```

```
=== KEYDB BUG REPORT START: Cut & paste starting from here ===
1719675:M 01 Jan 2020 23:26:00.740 # === ASSERTION FAILED ===
1719675:M 01 Jan 2020 23:26:00.740 # ==> db.cpp:1215 'itr != db->setexpire->end()' is not true
1719675:M 01 Jan 2020 23:26:00.740 # (forcing SIGSEGV to print the bug report.)
1719675:M 01 Jan 2020 23:26:00.740 # KeyDB 5.1.1 crashed by signal: 11
1719675:M 01 Jan 2020 23:26:00.740 # Crashed running the instruction at: 0x560ea6edfe1e
1719675:M 01 Jan 2020 23:26:00.740 # Accessing address: 0xffffffffffffffff
1719675:M 01 Jan 2020 23:26:00.740 # Failed assertion: itr != db->setexpire->end() (db.cpp:1215)

------ STACK TRACE ------
EIP:
/usr/local/bin/keydb-server 0.0.0.0:6379(_serverAssert+0x7e) [0x560ea6edfe1e]

Backtrace:
/usr/local/bin/keydb-server 0.0.0.0:6379(logStackTrace(ucontext*)+0x32) [0x560ea6ee1472]
/usr/local/bin/keydb-server 0.0.0.0:6379(sigsegvHandler(int, siginfo_t*, void*)+0xc8) [0x560ea6ee1b98]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x110e0) [0x7fdd655140e0]
/usr/local/bin/keydb-server 0.0.0.0:6379(_serverAssert+0x7e) [0x560ea6edfe1e]
/usr/local/bin/keydb-server 0.0.0.0:6379(removeExpireCore(redisDb*, redisObject*, dictEntry*)+0xdb7) [0x560ea6eacc67]
/usr/local/bin/keydb-server 0.0.0.0:6379(dbOverwriteCore(redisDb*, dictEntry*, redisObject*, redisObject*, bool, bool)+0x15d) [0x560ea6eae10d]
/usr/local/bin/keydb-server 0.0.0.0:6379(setKey(redisDb*, redisObject*, redisObject*)+0x5b) [0x560ea6eae3db]
/usr/local/bin/keydb-server 0.0.0.0:6379(setGenericCommand(client*, int, redisObject*, redisObject*, redisObject*, int, redisObject*, redisObject*)+0x87) [0x560ea6ec0a27]
/usr/local/bin/keydb-server 0.0.0.0:6379(setCommand(client*)+0x108) [0x560ea6ec0c88]
/usr/local/bin/keydb-server 0.0.0.0:6379(call(client*, int)+0xc7) [0x560ea6e8c007]
/usr/local/bin/keydb-server 0.0.0.0:6379(processCommand(client*, int)+0xac7) [0x560ea6e90817]
/usr/local/bin/keydb-server 0.0.0.0:6379(processCommandAndResetClient(client*, int)+0x21) [0x560ea6e99c01]
/usr/local/bin/keydb-server 0.0.0.0:6379(processInputBuffer(client*, int)+0x7a) [0x560ea6e9e1fa]
/usr/local/bin/keydb-server 0.0.0.0:6379(readQueryFromClient(aeEventLoop*, int, void*, int)+0x330) [0x560ea6ea0c00]
/usr/local/bin/keydb-server 0.0.0.0:6379(ProcessEventCore+0x176) [0x560ea6e84826]
/usr/local/bin/keydb-server 0.0.0.0:6379(aeProcessEvents+0x145) [0x560ea6e84a05]
/usr/local/bin/keydb-server 0.0.0.0:6379(aeMain+0x35) [0x560ea6e84d95]
/usr/local/bin/keydb-server 0.0.0.0:6379(workerThreadMain(void*)+0x97) [0x560ea6e89117]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x74a4) [0x7fdd6550a4a4]
/lib/x86_64-linux-gnu/libc.so.6(clone+0x3f) [0x7fdd6524cd0f]

------ INFO OUTPUT ------
```

**Comment by JohnSully** [comment]
Thanks, I've got a repro and will have a fix shortly.

**Comment by JohnSully** [comment]
This is fixed in change aea333bb

The expire is getting lost, and the assert is indicating that.  This will be in the next unstable docker and released shortly.
